pubsublite/internal/wire/settings.go

```diff
@@ -98,8 +102,8 @@ func validatePublishSettings(settings PublishSettings) error {
 	if settings.DelayThreshold <= 0 {
 		return errors.New("pubsublite: invalid publish settings. DelayThreshold duration must be > 0")
 	}
-	if settings.Timeout <= 0 {
-		return errors.New("pubsublite: invalid publish settings. Timeout duration must be > 0")
+	if settings.Timeout < minTimeout {
+		return errors.New("pubsublite: invalid publish settings. Timeout duration must be >= 2 minutes")
```

Comment:
I wonder if instead of erroring we just default to two minutes in these "error" cases if that really should be the min acceptable value. Still a behaviour change, but it will allow all old code to keep working without need of a code change. Thoughts?

Reply:
Thanks for the suggestion! I've clamped to a minimum of 2 minutes and logged a warning, in case users are relying on a low timeout. The warning says we will error in the future (say after 6 months).